compiler/rustc_serialize/src/opaque/mem_encoder.rs

```diff
@@ -31,12 +33,16 @@ impl MemEncoder {
     /// architectures support fixed-size writes up to 8 bytes with one instruction, so while this
     /// does in some sense do wasted work, we come out ahead.
     #[inline]
-    pub fn write_with<const N: usize>(&mut self, visitor: impl FnOnce(&mut [u8; N]) -> usize) {
+    pub fn write_with<const N: usize>(
+        &mut self,
+        visitor: impl FnOnce(&mut [MaybeUninit<u8>; N]) -> usize,
+    ) {
         self.data.reserve(N);
```

Comment:
I still think we should just not do this commit and instead use `resize` here on the `Vec` and use `last_chunk` to safely take out the array. I do not think this will have a performance impact in practice and it avoids the added maybeuninit complexity and the unsafe code

Reply:
`resize` + `last_chunk_mut` did have an performance impact (0.5%), but just initializing the array did not and still avoids `MaybeUninit`.

Reply:
makes sense